src/main/java/com/backend/webproject/dao/ProductDAO.java

```diff
@@ -36,14 +42,39 @@
         Map<String, Object> params = new HashMap<>();
         params.put("pName", pName);
         params.put("pCategoryID", pCategoryID);
-        List<Product> searchResult = jdbcTemplate.query(sql, params, ProductMapper);
+        List<Product> searchResult = jtemp.query(sql, params, ProductMapper);
 
         return searchResult;
     }
 
     public Product getProductById(int pID) {
-        Product product = jdbcTemplate.queryForObject("SELECT * FROM Product WHERE productID = :pID", new HashMap<String, Object>() {{put("pID", pID);}}, ProductMapper);
+        Product product = jtemp.queryForObject("SELECT * FROM Product WHERE productID = :pID", new HashMap<String, Object>() {{put("pID", pID);}}, ProductMapper);
         return product;
     }
 
+    public List<Product> getAllProducts() {
+        List<Product> allProducts = temp.query(
+                "SELECT * FROM Product ORDER BY productID ASC", ProductMapper);
+        return allProducts;
+    }
+
+    public void insertNewProduct(int pID, String pName, String pCompany, int pPrice, String pDescription, String pImagePath, int pCategoryID)
+	{
+		temp.update("INSERT INTO Product Values(?,?,?,?,?,?,?)", new Object[] {pID, pName, pCompany, pPrice, pDescription, pImagePath, pCategoryID});
+	}
+	
+    public void updateProduct(int pID, String pName, String pCompany, int pPrice, String pDescription, String pImagePath, int pCategoryID) {
+        temp.update("UPDATE Product SET productName = ?, productCompany = ?, productPrice = ?, productDescription = ?, productImagePath = ?, productCategoryId = ? where productID=?",
+                new Object[] { pName, pCompany, pPrice, pDescription, pImagePath, pCategoryID, pID });
+    }
+
+	public void deleteProduct(int pID)
+	{
+		temp.update("DELETE FROM Product WHERE productID = ?", new Object[] {pID});
+	}
```

Comment:
Return the update, as it returns the value of rows updated (in this cases just one). That way we can now if the database was updated or not.